### PR TITLE
TST: Move numpy-dev cron from daily to weekly

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to move numpy-dev cron job from daily to weekly. This is because we use the numpy-dev wheel which is only updated weekly, so I see little point in running it daily. We also run numpy-dev job for push events already.

This is a follow up of #10388 .

xref #11038

p.s. Cancelled CI because not needed.

### CODE REVIEW

File: .github/workflows/ci_cron.yml [hunk 1]
```diff
@@ -42,6 +32,6 @@ jobs:
       if: ${{ matrix.os == 'ubuntu-latest' }}
       run: sudo apt-get install language-pack-de tzdata
     - name: Install Python dependencies
-      run: python -m pip install --upgrade tox codecov
```
Comment (pllim) [inline .github/workflows/ci_cron.yml:45]: Also don't think we need `codecov` for cron jobs.

**Review by mhvk — APPROVED** [review]
Yes, makes total sense!

**Review by bsipocz — APPROVED** [review]

**Comment by pllim** [comment]
Thanks for the reviews!

**Comment by saimn** [comment]
Do we need to run pyinstaller daily ? :thinking:

**Comment by pllim** [comment]
> Do we need to run pyinstaller daily ?

Not sure. It is not run on the PR, so doesn't hurt to make sure changes for the day did not break it?

**Comment by saimn** [comment]
Well, it's just another packaging tool. Running it every day certainly helps to find which change could broke it, but I'm not sure if it's worth the extra computing cost ? (That said, if Github really wanted to saved some computation time on their CI, they should just stop running Actions on forks by default).

**Comment by pllim** [comment]
> if Github really wanted to saved some computation time on their CI

Or support skip ci, etc etc. Yeah.

**Comment by pllim** [comment]
Re: pyinstaller -- @saimn , if you want to pursue moving it to weekly, please open a new issue for discussion. Thanks!

**Pull request merged**